src/devices/badger_orion_me_enc.c

```diff
@@ -0,0 +1,149 @@
+                                              | xxLx |
+                                              +------+
+
+
+- LL: {8} Message length except CRC, mostly 0x14 = 20 bytes, to be confirmed.
+- II:{32} Fixed value, 0x00000000, could be reverse flow water counter ?
+- SS: {32} Serial Number, little-endian value
+- xx: Unknown, values look fixed and depends on the model, could be flags also, battery level too, to be guessed
+- L:{1} Leak
+- xx: other unknown values, flags, model, unit, battery low ? to be guessed.
+- RR: {32} Reading value, scale 10 gallon, little-endian value
+- DD: {32} Daily Reading Value, scale 10 gallon, little-endian value
+- FF:{8} Fixed value, always 0x10
+- CC:{16} CRC-16, poly 0x8005, init 0xFFFF, final XOR 0x0000, from previous 21 bytes.
+- TT: Trailing bits
+
+*/
+static int orion_me_enc_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t const preamble_pattern[] = { 0xaa, 0xaa, 0xec, 0x62, 0xec, 0x62};
+
+    uint8_t b[23];
+
+    if (bitbuffer->num_rows > 1) {
+        decoder_logf(decoder, 1, __func__, "Too many rows: %d", bitbuffer->num_rows);
+        return DECODE_FAIL_SANITY;
+    }
+    int msg_len = bitbuffer->bits_per_row[0];
+
+    if (msg_len > 290) {
+        decoder_logf(decoder, 1, __func__, "Packet too long: %d bits", msg_len);
+        return DECODE_ABORT_LENGTH;
+    }
+
+    int offset = bitbuffer_search(bitbuffer, 0, 0, preamble_pattern, sizeof(preamble_pattern) * 8);
+
+    if (offset >= msg_len) {
+        decoder_log(decoder, 1, __func__, "Sync word not found");
+        return DECODE_ABORT_EARLY;
+    }
+
+    if ((msg_len - offset ) < 184 ) {
+        decoder_logf(decoder, 1, __func__, "Packet too short: %d bits", msg_len);
+        return DECODE_ABORT_LENGTH;
+    }
+
+    offset += sizeof(preamble_pattern) * 8;
+    bitbuffer_extract_bytes(bitbuffer, 0, offset, b, 23 * 8);
+
+    // Unwhiten the data coded with IBM Whitening Algorithm LFSR.
+    ibm_whitening(b,23);
+
+    decoder_log_bitrow(decoder, 2, __func__, b, 23 * 8, "Unwhiten MSG");
+
+    if (crc16(b, 23, 0x8005, 0xffff)) {
+        decoder_log(decoder, 1, __func__, "CRC 16 do not match");
+        return DECODE_FAIL_MIC;
+    }
+
+    decoder_log_bitrow(decoder, 2, __func__, b, 23 * 8, "MSG");
+
+    // int msg_len   = b[0]; not used
+    int id            = b[8] << 24 | b[7] << 16 | b[6] << 8 | b[5];
+    int flags_1       = b[9] << 16 | b[10] << 8 | b[11];
+    int leaking       = (b[10] & 0x20) >> 5;
+    int reading_raw   = b[15] << 24 | b[14] << 16 | b[13] << 8 | b[12];
+    int daily_raw     = b[19] << 24 | b[18] << 16 | b[17] << 8 | b[18];
+    int flags_2       = b[20];
+    float volume_gal  = reading_raw * 0.1f; // scale or decimal could defer
+    float dly_vol_gal = daily_raw * 0.1f; // scale or decimal could defer
+
+    /* clang-format off */
+    data_t *data = data_make(
+            "model",             "",                    DATA_STRING, "Orion-MEENC",
```

Comment:
Nit: Technically this works for various other ORION models. We could hard-code the mapping from serial number ranges to model names. According to https://badgermeter.widen.net/content/vodetxkyxh/original?download=false&x.app=api:
* 30000000…59999999: ME or SE
* 60000000…69999999: Mobile M
* 70000000…89999999: Classic
* 110000000…119999999: LTE
* 120000000…129999999: LTE-M or LTE-MS
* 130000000…139999999: C or CS
* 140000000…148999999: HLA
* 149000000…149999999: HLC
* 150000000…159999999: HLB
* 160000000…169999999: HLD
* 170000000…179999999: HLFX
* 180000000…189999999: HLG